dips/dip-158.md

```diff
@@ -0,0 +1,805 @@
+
+#### GetInfoCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "GetInfoCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "GetInfoCommand",      
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98"   
+  }
+}
+```
+#### GetInfoCommand Response
+Field|Type|Required?|Description|
+|-|-|-|-|
+|payment_info|[PaymentInfoObject](#PaymentInfoObject)|Y|The minimal payment details required in order for the payer to decide whether the accept the payment or reject it.|
+
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+  "result": {
+    "_ObjectType": "PaymentInfoObject",
+    "payment_info": {      
+      "receiver": {
+        "address": "dm1pllhdmn9m42vcsamx24zrxgs3qqqpzg3ng32kvacapx5fl",         
+        "business_data": {
+          "name": "Acme",
+          "legal_name": "Acme Technologies Inc.",
+          "address": {
+            "city": "San Francisco",
+            "country": "US",
+            "line1": "1260 Market Street",
+            "line2": "Suite 450",
+            "postal_code": "94103",
+            "state": "CA"
+          }
+        }        
+      },
+      "action": {
+        "amount": "100000000",
+        "currency": "XUS",
+        "action": "charge",
+        "timestamp": "72322"
+      },
+      "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+      "description": "payment request description"     
+    }
+  }
+}
+```
+### InitChargeCommand
+| Command Type |Direction| Description | Request Data | Response Data |
+|-|-|-|-|-|
+|InitChargeCommand|Sender > Receiver|This functionality allows the wallet to init the payment process for payments of type `charge` based on the payment details. For example, this may occur after the customer approved the payment request|Reference ID ; Payer Data|Receiver Status|
+
+#### InitChargeCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|sender|[PaymentSenderObject](#paymentsenderobject)|Y|Minimal payer information required for the receiving VASP to enforce AML and perform risk checks|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "InitChargeCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "InitChargeCommand",
+    "sender": {
+      "address": "dm1pg9q5zs2pg9q5zs2pg9q5zs2pg9skzctpv9skzcg9kmwta",
```

Comment:
sender_address

if you don't like sender_address / receiver_address, I recommend account_address

Reply:
This rename would help with the double `address` field ambiguity mentioned in https://github.com/diem/dip/pull/166#discussion_r623164214